azure-pipelines.yml

```diff
@@ -0,0 +1,130 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
+stages:
+  - stage: Windows_nimi_bot
+    pool:
+      name: Users
+      demands:
+        - nimibot -equals True
+        - agent.os -equals Windows_NT
+    jobs:
+    - job: Windows_nimi_bot_Validate_Codegen
+      workspace:
+        clean: all
+      steps:
+        - task: PythonScript@0
+          displayName: Validate codegen
+          inputs:
+            scriptSource: inline
+            script: |
+              import os
+              import sys
+              import subprocess
+
+              # To verify code is generated
+              os.system('git config user.email "dummy@ni.com"')
+              os.system('git config user.name "dummy"')
+              commit_id_before_code_generator = os.system("git log -1 --format=%h")
+              os.system("tox -e codegen")
+              os.system('git commit -a -m "test"')
+              commit_id_after_code_generator = os.system("git log -1 --format=%h")
+              if commit_id_before_code_generator != commit_id_after_code_generator:
+                print("Please re-generate the codegen using command 'tox -e codegen'", file=sys.stderr)
+                sys.exit(-1)
+
+              # To verify code generated is not added in .gitignore list
+              output = subprocess.check_output(["git", "ls-files", "--others", "-i", "--exclude-standard"], shell=True).decode()
+              files_ignored = [ file_path for file_path in output.split("\n") if not file_path.startswith((".tox", 'commands.log', 'generated/driver_file_built', 'generated/printed_global_files', 'tools/__pycache__')) and file_path]
```

Comment:
Added ".tox", 'commands.log', 'generated/driver_file_built', 'generated/printed_global_files', 'tools/__pycache__' in the exception list
https://ni.visualstudio.com/Users/_build/results?buildId=1312033&view=logs&j=45c68899-b372-5281-6fc3-3df8e4c59a4a&t=7ed62bfa-b7ac-58d9-69f8-a6c96c9ca2af